tools/release.ps1

```diff
@@ -0,0 +1,168 @@
+param(
+  [parameter(Mandatory = $true)]
+  [String]$ReleaseVersion,
+
+  [parameter(Mandatory = $true)]
+  [String]$GitHubUsername,
+
+  [parameter(Mandatory = $true)]
+  [String]$GitHubToken
+)
+
+$ErrorActionPreference = 'Stop'
+
+# Adapted from https://www.herebedragons.io/powershell-create-github-release-with-artifact
+function Update-GitHubRelease {
+  param(
+    $VersionNumber,
+    $PreRelease,
+    $ReleaseNotes,
+    $ArtifactOutputDirectory,
+    $GitHubUsername,
+    $GitHubRepository,
+    $GitHubApiUsername,
+    $GitHubApiKey
+  )
+
+  $draft = $false
+
+  $auth = 'Basic ' + [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes($gitHubApiUsername + ':' + $gitHubApiKey));
+
+  # Github uses TLS 1.2
+  [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+
+  # Find existing release
+  $ReleaseDetails = $null
+  $releaseParams = @{
+    Uri         = "https://api.github.com/repos/$gitHubUsername/$gitHubRepository/releases/tags/$versionNumber";
+    Method      = 'GET';
+    ContentType = 'application/json';
+    Headers     = @{
+      Authorization = $auth;
+    }
+  }
+
+  try {
+    $ReleaseDetails = Invoke-RestMethod @releaseParams
+  }
+  catch {
+    # Release is missing, create it
+    $ReleaseDetails = $null
+  }
+
+  if ($ReleaseDetails -eq $null) {
+    Write-Host "Creating release $versionNumber"
+    # Create a release
+    $releaseData = @{
+      tag_name   = [string]::Format("{0}", $versionNumber);
+      name       = [string]::Format("{0}", $versionNumber);
+      body       = $releaseNotes;
+      draft      = $draft;
+      prerelease = $preRelease;
+    }
+    $releaseParams = @{
+      ContentType = 'application/json'
+      Uri         = "https://api.github.com/repos/$gitHubUsername/$gitHubRepository/releases";
+      Method      = 'POST';
+      Headers     = @{
+        Authorization = $auth;
+      }
+      Body        = (ConvertTo-Json $releaseData -Compress)
+    }
+    $ReleaseDetails = Invoke-RestMethod @releaseParams
+  }
+  else {
+    Write-Host "Updating release $versionNumber"
+    # Create a release
+    $releaseData = @{
+      tag_name   = [string]::Format("{0}", $versionNumber);
+      name       = [string]::Format("{0}", $versionNumber);
+      body       = $releaseNotes;
+      draft      = $draft;
+      prerelease = $preRelease;
+    }
+    $releaseParams = @{
+      ContentType = 'application/json'
+      Uri         = "https://api.github.com/repos/$gitHubUsername/$gitHubRepository/releases/$($ReleaseDetails.id)";
+      Method      = 'PATCH';
+      Headers     = @{
+        Authorization = $auth;
+      }
+      Body        = (ConvertTo-Json $releaseData -Compress)
+    }
+    $ReleaseDetails = Invoke-RestMethod @releaseParams
+  }
+
+  # Upload assets
+  $uploadUri = $ReleaseDetails | Select -ExpandProperty upload_url
+  $uploadUri = $uploadUri -creplace '\{\?name,label\}'
+
+  Get-ChildItem -Path $artifactOutputDirectory | % {
```

Comment:
Is this still true?

Reply:
Should we only be attaching `.vsix` with the correct version number?

Reply:
Ahh I see it's part of the instructions above.  Could we instead just search the working dir for output? or change the publish npm task to do that for us?

Reply:
Was trying to keep source as close as possible to your original https://github.com/lingua-pupuli/puppet-editor-services/commit/048f525b57ca3629c608d0e8312ef7ea08504738, so that eventually we can use it on both with a config file or something.

`vsce publish` does not leave a `vsix` around, so it will have to be package then publish.